server/src/main/resources/buildServerResources/editProjectConnectionVault.jsp

```diff
@@ -163,10 +163,19 @@
   $j('#OAuthConnectionDialog .popupSaveButtonsBlock .testConnectionButton').remove();
   $j("#testConnectionButton").appendTo($j('#OAuthConnectionDialog .popupSaveButtonsBlock')[0])
 
-  // here: ensure these are hidden if aws-iam-auth is enabled
+  $j(document).ready(function(){
+      let checked = $j('#aws-iam-auth').is(':checked')
+      $j("#approle_auth_endpoint").toggle(!checked);
+      $j("#approle_role_id").toggle(!checked);
+      $j("#approle_secret_id").toggle(!checked);
+  })
+
   $j('#aws-iam-auth').click(function() {
       $j("#approle_auth_endpoint").toggle(!this.checked);
+      $j("#approle_auth_endpoint input").val("")
```

Comment:
Please don't, it empties unsaved user input if someone accidentally clicked on that checkbox.
That should be done on server side